From 7c669783b1b3fab902ce34d7914b62617ed8b263 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Thu, 30 Jun 2022 22:35:05 +0100
Subject: [PATCH] dump/dump-data.sh: Test requires nbdkit 1.22

Ubuntu 20.04 has nbdkit 1.16 which lacks support for strings.  These
were added in nbdkit 1.22.

(cherry picked from commit a8fa05ffb8b85f41276ffb52498e4528c08e5f21)
---
 dump/dump-data.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dump/dump-data.sh b/dump/dump-data.sh
index 46e4d1e..11145b0 100755
--- a/dump/dump-data.sh
+++ b/dump/dump-data.sh
@@ -25,6 +25,10 @@ requires nbdkit --version
 requires nbdkit data --dump-plugin
 requires nbdkit -U - null --run 'test "$uri" != ""'
 
+# This test requires nbdkit >= 1.22.
+minor=$( nbdkit --dump-config | grep ^version_minor | cut -d= -f2 )
+requires test $minor -ge 22
+
 output=dump-data.out
 rm -f $output
 cleanup_fn rm -f $output
-- 
2.31.1

